core/discord/src/main/kotlin/kord/interaction/KordApplicationCommandScope.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022 AndroidDev Discord Dev Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kord.interaction
+
+import dev.kord.gateway.Gateway
+import dev.kord.gateway.InteractionCreate
+import dev.kord.rest.service.RestClient
+import interaction.ApplicationCommandScope
+import interaction.ChatInputCommandBuilder
+import interaction.InteractionScope
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.filterIsInstance
+import kotlinx.coroutines.launch
+
+internal class KordApplicationCommandScope(
+    private val restClient: RestClient,
+    private val gateway: Gateway,
+    private val scope: CoroutineScope,
+) : ApplicationCommandScope {
+    override suspend fun registerGlobalChatInputCommand(
```

Comment:
I don't think this needs to be a suspend function

Reply:
Registering the command makes a web request, which is suspending as per Kord's API.

I do like that the way it's currently set up will suspend until the command is registered, but I'm happy to make it non-suspending and just `launch` the whole thing if you'd rather

Reply:
those are 2 completely different things tho'

one is listening to commands and publish them to observer, the other is the network call.

Unless I'm missing something